### PR TITLE
Add unit tests for is_weak_hash_algo in ssl module

### PR DESCRIPTION
**Pull request by Raavi29**
## Proposed change

Adds unit tests for the `is_weak_hash_algo()` function in `nettacker/core/lib/ssl.py`

This function had no dedicated unit tests. These tests verify:
- Weak algorithms are correctly detected: sha1, md5, md2, md4
- Case insensitivity works (SHA1, MD5 uppercase inputs caught correctly)
- Safe algorithms are not flagged: sha256, sha512, sha384
- Edge case handled: empty string, unrecognised algorithm 

All 11 tests passing locally via `python -m pytest tests\test_ssl.py -v`.
This is part of improving overall test coverage for Nettacker.

## Type of change

- [ ] New core framework functionality
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Code refactoring without any functionality changes
- [ ] New or existing module/payload change
- [ ] Documentation/localization improvement
- [x] Test coverage improvement
- [ ] Dependency upgrade
- [ ] Other improvement (best practice, cleanup, optimization, etc)

## Checklist

- [ ] I've followed the [contributing guidelines][contributing-guidelines]
- [ ] I have **digitally signed** all my commits in this PR
- [ ] I've run `make pre-commit` and confirm it didn't generate any warnings/changes
- [ ] I've run `make test`, I confirm all tests passed locally
- [ ] I've added/updated any relevant documentation in the `docs/` folder 
- [ ] I've linked this PR with an open issue
- [ ] I've tested and verified that my code works as intended and resolves the issue as described
- [ ] I have attached screenshots demonstrating my code works as intended
- [x] I've checked all other open PRs to avoid submitting duplicate work
- [x] I confirm that the code and comments in this PR are not direct unreviewed outputs of AI
- [x] I confirm that I am the Sole Responsible Author for every line of code, comment, and design decision

[contributing-guidelines]: https://nettacker.readthedocs.io/en/latest/Developers/

### CODE REVIEW

File: tests/test_ssl.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
+# tests/test_ssl.py
+# Tests for nettacker/core/lib/ssl.py
```
[the remark this answers is not in the document]
Comment (Raavi29) [reply tests/test_ssl.py]: Thanks for the feedback. Moved the test file to tests/core/lib/test_ssl.py to mirror the package layout.

File: tests/core/lib/test_ssl.py [hunk 2]
```diff
@@ -1,574 +1,58 @@
-import ssl
-from unittest.mock import patch
+# tests/core/lib/test_ssl.py
+# Tests for nettacker/core/lib/ssl.py
+# Author: Parneet Kaur
+# GSoC 2026 - OWASP Nettacker
```
[the remark this answers is not in the document]
Comment (Raavi29) [reply tests/core/lib/test_ssl.py:4]: Fixed

File: tests/core/lib/test_ssl.py [hunk 3]
```diff
@@ -1,574 +1,58 @@
-import ssl
-from unittest.mock import patch
+# tests/test_ssl.py
+# Tests for nettacker/core/lib/ssl.py
+# Author: Parneet Kaur
+# GSoC 2026 - OWASP Nettacker
```
[the remark this answers is not in the document]
Comment (Raavi29) [reply tests/core/lib/test_ssl.py]: fixed

File: tests/core/lib/test_ssl.py [hunk 4]
```diff
@@ -1,574 +1,58 @@
...
 
 import pytest
+from nettacker.core.lib.ssl import is_weak_hash_algo, get_cert_info
```
[the remark this answers is not in the document]
Comment (Raavi29) [reply tests/core/lib/test_ssl.py]: fixed

File: tests/core/lib/test_ssl.py [hunk 5]
```diff
@@ -1,574 +1,58 @@
...
-        )
-
-        assert result == []
-
-    def test_response_conditions_matched_none_response(self, ssl_engine, substeps):
-        result = ssl_engine.response_conditions_matched(substeps.ssl_weak_version_vuln, None)
-
-        assert result == []
+    def test_random_string_is_not_weak(self):
+        assert is_weak_hash_algo("someRandomAlgorithm") is False
```
[the remark this answers is not in the document]
Comment (Raavi29) [reply tests/core/lib/test_ssl.py:58]: get_cert_info, is_weak_ssl_version, is_weak_cipher_suite, and create_socket_connection were not covered in the previous version of this PR either - this PR only ever targeted is_weak_hash_algo. Tests for the remaining functions require network mocking and are planned as follow-up contributions. The unused import of get_cert_info was removed per the previous review suggestion.

File: tests/test_ip.py [hunk 6]
```diff
@@ -0,0 +1,296 @@
...
+from nettacker.core.ip import (
+    generate_ip_range,
+    get_ip_range,
+    is_single_ipv4,
+    is_ipv4_range,
+    is_ipv4_cidr,
+    is_single_ipv6,
+    is_ipv6_range,
+    is_ipv6_cidr,
+)
```
[the remark this answers is not in the document]
Comment (Raavi29) [reply tests/test_ip.py:12]: Moved to tests/core/test_ip.py to mirror the package layout.

**Comment by Raavi29** [comment]
Closing in favour of a cleaner PR that appends to the existing test suite rather than replacing it. See replacement PR for the same tests correctly added.
